cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -122,6 +122,34 @@ CAST_UNARY(castFLOAT4, float64, float32)
 
 #undef CAST_UNARY
 
+// cast float types to int types.
+#define CAST_INT_FLOAT(NAME, IN_TYPE, OUT_TYPE)       \
+  FORCE_INLINE                                        \
+  gdv_##OUT_TYPE NAME##_##IN_TYPE(gdv_##IN_TYPE in) { \
```

Comment:
why not just do a cast to int with static_cast

Reply:
Because, the cast_int_float function needs add on to the out int number if the decimal part of the number to be cast is bigger than 0,4, eg. 5.6666 should end up as 6 after cast.

Reply:
why not use the round() function that is already added

Reply:
Done, sorry I forgot about it